replay2/memory_remapper/tests/memory_remapper_tests.cc

```diff
@@ -0,0 +1,184 @@
+		ASSERT_EQ(replayAddress.charPtr()[0], i % 256);
+	}
+}
+
+void ASSERT_CONST_REPLAY_ADDRESS(const MemoryRemapper& remapper, const CaptureAddress& captureAddress, const ReplayAddress& replayAddress, char value, size_t length) {
+	for(size_t i = 0; i < length; ++i) {
+		ReplayAddress replayAddress = remapper.RemapCaptureAddress(captureAddress.offsetByBytes(i));
+		ASSERT_EQ(replayAddress.charPtr()[0], value);
+	}
+}
+
+TEST(MemoryRemapperTests, SimpleMapping) {
+
+	const size_t size = 128;
+
+	char* rawCapturePtr = new char[size];
+	CaptureAddress captureAddress(rawCapturePtr);
+
+	MemoryRemapper remapper;
+	const MemoryObservation captureObservation(captureAddress, std::make_shared<ModResourceGenerator>(size));
+
+	const ReplayAddress replayAddress = remapper.AddMapping(captureObservation);
+	EXPECT_NE(replayAddress.charPtr(), nullptr);
+	ASSERT_MOD_REPLAY_ADDRESS(remapper, captureAddress, replayAddress, size);
+
+	EXPECT_NO_THROW(remapper.RemoveMapping(captureAddress));
+	EXPECT_THROW(remapper.RemapCaptureAddress(captureAddress), MemoryRemapper::AddressNotMappedException);
+}
+
+TEST(MemoryRemapperTests, UnknownMapping) {
+
+	const size_t size = 128;
+
+	char* rawCapturePtr = new char[size];
+	CaptureAddress captureAddress(rawCapturePtr);
+
+	MemoryRemapper remapper;
+
+	EXPECT_THROW(remapper.RemapCaptureAddress(captureAddress), MemoryRemapper::AddressNotMappedException);
+
+	EXPECT_THROW(remapper.RemoveMapping(captureAddress), MemoryRemapper::AddressNotMappedException);
+	EXPECT_THROW(remapper.RemapCaptureAddress(captureAddress), MemoryRemapper::AddressNotMappedException);
+}
+
+TEST(MemoryRemapperTests, MultipleMappings) {
+
+	std::vector<ReplayAddress> replayAddresses;
+	std::vector<CaptureAddress> captureAddresses;
+
+	MemoryRemapper remapper;
+
+	for(int i = 0; i < 64; ++i) {
+
+		const size_t size = i * 2;
+
+		char* rawCapturePtr = new char[size];
+		CaptureAddress captureAddress(rawCapturePtr);
+		captureAddresses.push_back(captureAddress);
+
+		const MemoryObservation captureObservation(captureAddress, std::make_shared<ConstResourceGenerator>(i, size));
+
+		const ReplayAddress replayAddress = remapper.AddMapping(captureObservation);
+		replayAddresses.push_back(replayAddress);
+	}
+
+	for(int i = 0; i < 64; ++i) {
+		const size_t size = i * 2;
+		EXPECT_NE(replayAddresses[i].charPtr(), nullptr);
+		ASSERT_CONST_REPLAY_ADDRESS(remapper, captureAddresses[i], replayAddresses[i], i, size);
+		EXPECT_NO_THROW(remapper.RemoveMapping(captureAddresses[i]));
+		EXPECT_THROW(remapper.RemapCaptureAddress(captureAddresses[i]), MemoryRemapper::AddressNotMappedException);
+	}
+}
+
+TEST(MemoryRemapperTests, MappingCollision) {
+
+	const size_t offset = 31;
+
+	const size_t sizeA = 128;
+	char* rawCapturePtrA = new char[sizeA];
+	CaptureAddress captureAddressA(rawCapturePtrA);
+
+	const size_t sizeB = sizeA -offset;
+	char* rawCapturePtrB = rawCapturePtrA +31;
```

Comment:
Nit: Offset instead of 31

Reply:
Fixed

Reply:
Hmm.. it's still 31 but maybe you forgot to push?

Reply:
I haven't pushed any of the updates yet sorry!